src/shared/elf-util.c

```diff
@@ -538,6 +538,10 @@ static int module_callback(Dwfl_Module *mod, void **userdata, const char *name,
                 if (!program_header || program_header->p_type != PT_LOAD)
                         continue;
 
+                /* This PT_LOAD segment doesn't contain the start address, so it can't be the module we are looking for. */
+                if (start < program_header->p_vaddr || start >= program_header->p_vaddr + program_header->p_memsz)
```

Comment:
do we need to be afraid of overflows on that addition?

Reply:
@Romain-Geissler-1A I missed this open question - can you double check?

Reply:
ahah that's funny, now it shows your answer before my question, but the timestamps are inverted :-)

> I guess it all depends on how well we trust the core dump file.

In general we consider core files we analyze to be untrusted, so it's better safe than sorry.

Reply:
Ah so I have to prevent this overflow indeed.

See #28128.